drivers/CMakeLists.txt

```diff
@@ -23,16 +22,17 @@ add_subdirectory_if_kconfig(pinmux)
 add_subdirectory_if_kconfig(pwm)
 add_subdirectory_if_kconfig(rtc)
 add_subdirectory_if_kconfig(sensor)
-add_subdirectory_if_kconfig(serial)
 add_subdirectory_if_kconfig(spi)
 add_subdirectory_if_kconfig(usb)
 add_subdirectory_if_kconfig(watchdog)
 add_subdirectory_if_kconfig(wifi)
 add_subdirectory_if_kconfig(can)
 
+add_subdirectory_ifdef(CONFIG_FLASH_HAS_DRIVER_ENABLED flash)
+add_subdirectory_ifdef(CONFIG_SERIAL_HAS_DRIVER serial)
 add_subdirectory_ifdef(CONFIG_AIO_COMPARATOR   aio)
 add_subdirectory_ifdef(CONFIG_BT               bluetooth)
 add_subdirectory_ifdef(CONFIG_NETWORKING       net)
 add_subdirectory_ifdef(CONFIG_NET_L2_ETHERNET  ethernet)
-add_subdirectory_ifdef(CONFIG_ENTROPY_GENERATOR entropy)
+add_subdirectory_ifdef(CONFIG_ENTROPY_HAS_DRIVER entropy)
```

Comment:
Will changing these semantics cause entropy drivers, that used to be pruned, to now be included into applications?

Is there any way of proving yes / no. E.g. by running sanitycheck and comparing sizes before and after.

The reason I suspect there is a risk of this is that this build script used to say:

"Include the entropy driver when the entropy generator is enabled."

But now it says:

"Include the entropy driver when the entropy generator is supported."

Reply:
@SebastianBoe I understand the issue here. But this Kconfig "entropy has driver" will only be enabled if entropy_generator is enabled. So we will maintain the same subdirectory include condition as before.

Reply:
I see, then this is safe.